src/m365/spo/commands/page/page-control-list.ts

```diff
@@ -31,39 +32,44 @@ class SpoPageControlListCommand extends SpoCommand {
   }
 
   public commandAction(logger: Logger, args: CommandArgs, cb: (err?: any) => void): void {
-    Page
-      .getPage(args.options.name, args.options.webUrl, logger, this.debug, this.verbose)
-      .then((clientSidePage: ClientSidePage): void => {
-        let controls: ClientSidePart[] = [];
-        clientSidePage.sections.forEach(s => {
-          s.columns.forEach(c => {
-            controls = controls.concat(c.controls);
-          });
-        });
-        // remove the column property to be able to serialize the array to JSON
-        controls.forEach(c => delete c.column);
+    let pageName: string = args.options.name;
+    if (args.options.name.indexOf('.aspx') < 0) {
+      pageName += '.aspx';
+    }
 
-        // remove the dynamicDataValues and dynamicDataPaths properties if they are null
-        controls.forEach(c => {
-          if (!c.dynamicDataPaths) {
-            delete c.dynamicDataPaths
-          }
-          if (!c.dynamicDataValues) {
-            delete c.dynamicDataValues
-          }
-        });
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/SitePages/Pages/GetByUrl('sitepages/${encodeURIComponent(pageName)}')`,
+      headers: {
+        'content-type': 'application/json;charset=utf-8',
+        accept: 'application/json;odata=nometadata'
+      },
+      responseType: 'json'
+    };
 
-        controls.forEach(c => {
-          (c as any).type = SpoPageControlListCommand.getControlTypeDisplayName((c as any).controlType);
+    request
+      .get<ClientSidePageProperties>(requestOptions)
+      .then((clientSidePage: ClientSidePageProperties): void => {
+        const canvasData: Control[] = JSON.parse(clientSidePage.CanvasContent1);
+        let controls: any[] = canvasData.filter(c => c.position).map(c => {
+          return {
+            controlType: c.controlType,
+            order: c.position.sectionIndex,
+            id: c.id,
+            controlData: {
+              ...c
+            },
+            ...c.webPartData || {},
+            type: this.getControlTypeDisplayName(c.controlType || 0)
+          };
         });
-        // drop the information about original classes from clientsidepages.ts
+        
         logger.log(JSON.parse(JSON.stringify(controls)));
 
         cb();
       }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
   }
 
-  private static getControlTypeDisplayName(controlType: number): string {
+  private getControlTypeDisplayName(controlType: number): string {
```

Comment:
I don't think we need to change it to an instance method, as it doesn't use any information from the class.

Reply:
This method is now also available in another file, so we can remove it in a future update.